### ISSUE TITLE
frame #4: <unknown function> + 0xdbbf4 (0x7eff47d5ebf4 in /home/shraban/anaconda3/envs/rajnish/bin/../lib/libstdc++.so.6) frame #5: <unknown function> + 0x8609 (0x7eff493c1609 in /lib/x86_64-linux-gnu/libpthread.so.0) frame #6: clone + 0x43 (0x7eff4918c353 in /lib/x86_64-linux-gnu/libc.so.6)  Error during training: process 0 terminated with signal SIGABRT

### ISSUE THREAD

**Issue by rajnish159** [IBM/Grapher#24]
i trained the model, but i found this type of error, please help me.

by using this command :-
$  bash train_gen.sh

**Comment by echo840** [comment]
Same question. How did you solve this problem?

**Comment by rajnish159** [comment]
i solved this error

**Comment by echo840** [comment]
![image](https://github.com/IBM/Grapher/assets/87795401/407b2d7b-e710-4be8-a8e3-38092e204fee)
I have a similar problem. Could you please tell me how to solve it? Thank you very much.

**Comment by rajnish159** [comment]
i used multiple GPU's ,

**Comment by rajnish159** [comment]
from data.dataset import GraphDataModule
from pytorch_lightning import loggers as pl_loggers
import argparse
from model.litgrapher import LitGrapher
import pytorch_lightning as pl
from transformers import T5Tokenizer, T5ForConditionalGeneration
import os
from pytorch_lightning.callbacks import ModelCheckpoint, RichProgressBar
from misc.utils import decode_graph

os.environ["CUDA_VISIBLE_DEVICES"] = "0,1,2,3,4,5,6"

def main(args):

    args.eval_dir = os.path.join(args.default_root_dir, args.dataset + '_version_' + args.version)
    args.checkpoint_dir = os.path.join(args.eval_dir, 'checkpoints')

    os.makedirs(args.checkpoint_dir, exist_ok=True)
    os.makedirs(os.path.join(args.eval_dir, 'valid'), exist_ok=True)
    os.makedirs(os.path.join(args.eval_dir, 'test'), exist_ok=True)

    TB = pl_loggers.TensorBoardLogger(save_dir=args.default_root_dir, name='', version=args.dataset + '_version_' + args.version, default_hp_metric=False)

    if args.checkpoint_model_id < 0:
        checkpoint_model_path = os.path.join(args.checkpoint_dir, 'last.ckpt')
    else:
        checkpoint_model_path = os.path.join(args.checkpoint_dir, f"model-step={args.checkpoint_model_id}.ckpt")

    if args.run == 'train':

        dm = GraphDataModule(tokenizer_class=T5Tokenizer,
                             tokenizer_name=args.pretrained_model,
                             cache_dir=args.cache_dir,
                             data_path=args.data_path,
                             dataset=args.dataset,
                             batch_size=args.batch_size,
                             num_data_workers=args.num_data_workers,
                             max_nodes=args.max_nodes,
                             max_edges=args.max_edges,
                             edges_as_classes=args.edges_as_classes)

        dm.prepare_data()
        dm.setup(stage='fit')
        dm.setup(stage='validate')

        checkpoint_callback = ModelCheckpoint(
            dirpath=args.checkpoint_dir,
            filename='model-{step}',
            save_last=True,
            save_top_k=-1,
            every_n_train_steps=args.checkpoint_step_frequency,
        )

        grapher = LitGrapher(transformer_class=T5ForConditionalGeneration,
                             transformer_name=args.pretrained_model,
                             tokenizer=dm.tokenizer,
                             cache_dir=args.cache_dir,
                             max_nodes=args.max_nodes,
                             max_edges=args.max_edges,
                             edges_as_classes=args.edges_as_classes,
                             default_seq_len_edge=args.default_seq_len_edge,
                             num_classes=len(dm.dataset_train.edge_classes),
                             dropout_rate=args.dropout_rate,
                             num_layers=args.num_layers,
                             vocab_size=len(dm.tokenizer.get_vocab()),
                             bos_token_id=dm.tokenizer.pad_token_id,
                             eos_token_id=dm.tokenizer.eos_token_id,
                             nonode_id=dm.tokenizer.convert_tokens_to_ids('__no_node__'),
                             noedge_id=dm.tokenizer.convert_tokens_to_ids('__no_edge__'),
                             node_sep_id=dm.tokenizer.convert_tokens_to_ids('__node_sep__'),
                             noedge_cl=len(dm.dataset_train.edge_classes) - 1,
                             edge_classes=dm.dataset_train.edge_classes,
                             focal_loss_gamma=args.focal_loss_gamma,
                             eval_dir=args.eval_dir,
                             lr=args.lr)

        if not os.path.exists(checkpoint_model_path):
            checkpoint_model_path = None

        trainer = pl.Trainer.from_argparse_args(args,
                                                logger=TB,
                                                callbacks=[checkpoint_callback, RichProgressBar(10)])

        dm.setup(stage='validate')

        trainer.fit(model=grapher, datamodule=dm, ckpt_path=checkpoint_model_path)
        
    elif args.run == 'test':

        assert os.path.exists(checkpoint_model_path), 'Provided checkpoint does not exists, cannot run the test'

        grapher = LitGrapher.load_from_checkpoint(checkpoint_path=checkpoint_model_path)

        dm = GraphDataModule(tokenizer_class=T5Tokenizer,
                             tokenizer_name=grapher.transformer_name,
                             cache_dir=grapher.cache_dir,
                             data_path=args.data_path,
                             dataset=args.dataset,
                             batch_size=args.batch_size,
                             num_data_workers=args.num_data_workers,
                             max_nodes=grapher.max_nodes,
                             max_edges=grapher.max_edges,
                             edges_as_classes=grapher.edges_as_classes)

        dm.setup(stage='test')

        trainer = pl.Trainer.from_argparse_args(args, logger=TB)

        trainer.test(grapher, datamodule=dm)

    else: # single inference

        assert os.path.exists(checkpoint_model_path), 'Provided checkpoint does not exists, cannot do inference'

        grapher = LitGrapher.load_from_checkpoint(checkpoint_path=checkpoint_model_path)

        tokenizer = T5Tokenizer.from_pretrained(grapher.transformer_name, cache_dir=grapher.cache_dir)
        tokenizer.add_tokens('__no_node__')
        tokenizer.add_tokens('__no_edge__')
        tokenizer.add_tokens('__node_sep__')
        
        text_tok = tokenizer([args.inference_input_text],
                             add_special_tokens=True,
                             padding=True,
                             return_tensors='pt')

        text_input_ids, mask = text_tok['input_ids'], text_tok['attention_mask']

        _, seq_nodes, _, seq_edges = grapher.model.sample(text_input_ids, mask)

        dec_graph = decode_graph(tokenizer, grapher.edge_classes, seq_nodes, seq_edges, grapher.edges_as_classes,
                                grapher.node_sep_id, grapher.max_nodes, grapher.noedge_cl, grapher.noedge_id,
                                grapher.bos_token_id, grapher.eos_token_id)
        
        graph_str = ['-->'.join(tri) for tri in dec_graph[0]]
        
        print(f'Generated Graph: {graph_str}')
        
    
if __name__ == "__main__":
    
    # Parsing arguments
    parser = argparse.ArgumentParser(description='Arguments')

    parser.add_argument("--dataset", type=str, default='webnlg')
    parser.add_argument("--run", type=str, default='train')
    parser.add_argument('--pretrained_model', type=str, default='t5-base')
    parser.add_argument('--version', type=str, default='0')
    parser.add_argument('--data_path', type=str, default='')
    parser.add_argument('--cache_dir', type=str, default='cache')
    parser.add_argument('--num_data_workers', type=int, default=3)
    parser.add_argument('--checkpoint_step_frequency', type=int, default=-1)
    parser.add_argument('--checkpoint_model_id', type=int, default=-1)
    parser.add_argument('--max_nodes', type=int, default=8)
    parser.add_argument('--max_edges', type=int, default=7)
    parser.add_argument('--default_seq_len_node', type=int, default=20)
    parser.add_argument('--default_seq_len_edge', type=int, default=20)
    parser.add_argument('--edges_as_classes', type=int, default=0)
    parser.add_argument('--batch_size', type=int, default=4)
    parser.add_argument('--lr', default=1e-5, type=float)
    parser.add_argument("--focal_loss_gamma", type=float, default=0.0)
    parser.add_argument("--dropout_rate", type=float, default=0.5)
    parser.add_argument("--num_layers", type=int, default=1)
    parser.add_argument("--eval_dump_only", type=int, default=0)
    parser.add_argument("--inference_input_text", type=str,
                        default='Danielle Harris had a main role in Super Capers, a 98 minute long movie.')

    parser = pl.Trainer.add_argparse_args(parser)

    args = parser.parse_args()

    main(args)
    
    please change some things, in main.py

**Comment by rajnish159** [comment]
#!/usr/bin/env bash
cd ../
python main.py    --version 2\
                  --default_root_dir output \
                  --run train \
                  --max_epochs 20 \
                  --accelerator gpu \
                  --num_nodes 1 \
                  --num_data_workers 4 \
                  --lr 1e-4 \
                  --batch_size 4 \
                  --num_sanity_val_steps 0 \
                  --fast_dev_run 0 \
                  --overfit_batches 0 \
                  --limit_train_batches 1.0 \
                  --limit_val_batches 1.0 \
                  --limit_test_batches 1.0 \
                  --accumulate_grad_batches 4 \
                  --detect_anomaly True \
                  --data_path webnlg-dataset/release_v3.0/en \
                  --log_every_n_steps 100 \
                  --val_check_interval 1000 \
                  --checkpoint_step_frequency 1000 \
                  --focal_loss_gamma 3 \
                  --dropout_rate 0.5 \
                  --num_layers 2 \
                  --edges_as_classes 0 \
                  --checkpoint_model_id -1 \


train_gen.sh file
